python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,380 @@
+"""Torch Module for PGExplainer"""
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import Adam
+
+__all__ = ["PGExplainer"]
+
+
+class PGExplainer(nn.Module):
+    r"""PGExplainer from `Parameterized Explainer for Graph Neural Network
+    <https://arxiv.org/pdf/2011.04573>`
+
+    PGExplainer adopts a deep neural network to parameterize the generation
+    process of explanations, which enables it to explain multiple instances
+    collectively. PGExplainer models the underlying structure as edge
+    distributions, from which the explanatory graph is sampled.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat, embed, edge_weight)`.
+        * The output of its forward function is the logits if embed=False else
+        return the graph embedding.
+    num_features : int
+        Number of input feature for the explanation network.
+    epochs : int, optional
+        Number of epochs to train the explanation network. Default: 10.
+    learning_rate : float, optional
+        Learning rate to train the explanation network. Default: 0.01.
+    coff_size : float, optional
+        Size regularization to constrain the explanation size. Default: 0.01.
+    coff_ent : float, optional
```

Comment:
```suggestion
    coff_connect : float, optional
```
to be consistent with the paper